include/fiction/technology/cell_ports.hpp

```diff
@@ -299,15 +299,53 @@ struct formatter<fiction::port_direction>
     template <typename FormatContext>
     auto format(const fiction::port_direction& p, FormatContext& ctx) const
     {
-        return format_to(ctx.out(), p.dir == fiction::port_direction::NORTH      ? "N" :
-                                    p.dir == fiction::port_direction::NORTH_EAST ? "NE" :
-                                    p.dir == fiction::port_direction::EAST       ? "E" :
-                                    p.dir == fiction::port_direction::SOUTH_EAST ? "SE" :
-                                    p.dir == fiction::port_direction::SOUTH      ? "S" :
-                                    p.dir == fiction::port_direction::SOUTH_WEST ? "SW" :
-                                    p.dir == fiction::port_direction::WEST       ? "W" :
-                                    p.dir == fiction::port_direction::NORTH_WEST ? "NW" :
-                                                                                   "?");
+        const auto* dir = "?";
+
+        switch (p.dir)
```

Comment:
May indeed, but does here. It's actually a best practice to cover all cases and leave out the `default` branch so that when you add elements to the `enum class`, the compiler will warn you for all affected `switch` statements. If you add a `default`, no warning will be triggered.